### PR TITLE
Update book

### PR DESCRIPTION
**Pull request by Schaeff**
This is an update of the book for the coming release. It focuses on removing content and relying on the CLI `--help` and stdlib source comments until we have a reliable way to keep the book and the implementation in sync, which we've struggled with in the past.

- Remove CLI docs and rely on `--help`
- Remove stdlib docs and rely on source
- Improve stdlib docs
- Remove useless stdlib components
- Typos

### CODE REVIEW

**Comment by fulldecent** [comment]
Thank you, please let me know that status on this.

I was just about to get started on updating documentation and comprehensive CLI docs was the first task I was looking at. Glad I saw this PR in the meantime.

File: zokrates_book/src/toolbox/stdlib.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+### Hashes
+
+#### SHA256
+We provide an implementation of the SHA256 function from the SHA-2 family of secure hash functions [^1]. The hash functions of the SHA-2 family are considered to be pseudorandom.
+
+SHA256 is available in Ethereum as a pre-compiled contract and thus a hash function that is cheap to evaluate in the EVM. However, the implementation inside a circuit is comparatively expensive, as it is defined for binary in- and outputs and heavily relies on bit manipulation.
+
+
+#### Pedersen Hashes
+The pedersen hash function is derived from a commitment scheme published by Pedersen [^2].
```
Comment (JacobEberhardt) [inline zokrates_book/src/toolbox/stdlib.md:16]:
```suggestion
The pedersen hash function is inspired by a commitment scheme published by Pedersen [^2].
```

File: zokrates_book/src/toolbox/stdlib.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+This hash function’s security is based the discrete logarithm problem.
```
Comment (JacobEberhardt) [inline zokrates_book/src/toolbox/stdlib.md:17]:
```suggestion
This hash function’s security is based on the discrete logarithm problem. 
```

File: zokrates_book/src/toolbox/stdlib.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+This hash function’s security is based the discrete logarithm problem. 
+Pedersen-hashes cannot be assumed to be pseudorandom and should therefore not be used where a hash function serves as a random oracle.
+
+In the EVM, operations on the BabyJubJub curve are not natively supported. Therefore, Pedersen hashes are expensive to evaluate on-chain and should be avoided.
+
+By definition, the Pedersen hash function has a fixed-length binary input and outputs a group element, i.e., a point on the BabyJubJub elliptic curve in our case.
+
+#### MiMC
+The MiMC hash function was designed by using the MiMC-Feistel permutation [^3] over a prime field in a sponge construction [^4] to arrive at a secure and efficiently provable hash function.
+The construction is based on established hash function design principles from symmetric cryptography but is still novel and under review by the cryptography community. MiMC hashes are considered to be pseudorandom.
```
Comment (JacobEberhardt) [inline zokrates_book/src/toolbox/stdlib.md:26]:
```suggestion
The construction is based on established hash function design principles from symmetric cryptography but is still novel and should thus be used cautiously. MiMC hashes are considered to be pseudorandom.
```

File: zokrates_book/src/toolbox/stdlib.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+Due to its native use of prime field arithmetics, MiMC hash functions are efficient in circuits. At the same time, they can be evaluated by the EVM with comparatively little overhead.
+
+The MiMC hash function maps from field elements to field elements; applying the function to its output again does not introduce overhead.
```
Comment (JacobEberhardt) [inline zokrates_book/src/toolbox/stdlib.md:30]:
```suggestion
The MiMC hash function maps from field elements to field elements; applying the function to its output again does not introduce overhead for packing/unpacking.
```

**Pull request merged**